### DISCUSSION TITLE
search doesn't work immediately after indexing completes

### DISCUSSION THREAD

**Discussion by jkjha** [oracle/opengrok#4122] [Q&A]
We found some cases where search on UI doesn't work just after (even after 1 hour) completing the indexing. The index directory doesn't have any dirty file, so that means index is clean but I see some files such as "segment_b" along with other index files. During the next indexing cycle, it creates another file such as "segment_c" ...

Can you please help me understand what does that mean? Also, what could be a reason of search not working after the index? I can browse the files along with History, annotation etc all working fine but search doesn't work.

I see similar issue reported here where deleting the index solved the issue, but before doing that checking if thats only the workaround and what could cause this?

https://github.com/oracle/opengrok/issues/3590

Some more details:

Opengrok version: 1.7.41 on docker (built from source with following changes in docker/sync.yml file)

```
- command: [opengrok-reindex-project,
    -J=-XX:-UseGCOverheadLimit, -J=-Xmx36g, -J=-server, --jar, /opengrok/lib/opengrok.jar,
    -t, /opengrok/etc/logging.properties.template, -p, '%PROJ%', -d, /opengrok/log/%PROJECT%,
    -P, '%PROJECT%', -U, '%URL%', --, --renamedHistory, 'off', -r, dirbased, -G, -m, '4096', -c,
    /usr/local/bin/ctags, -U, '%URL%', -o, /opengrok/etc/ctags.config,
    -H, '%PROJECT%']
  limits: {RLIMIT_NOFILE: 5020}
```

Project size: 2.7GB

**Comment by jkjha** [comment]
After next indexer run, it created new file something `segments_4` but after that now search is working (almost after 2 successful indexer run). So, just checking if thats expected?

**Reply by vladak** [reply]
What does it mean exactly that search is not working ?

**Reply by jkjha** [reply]
it returns "Your search full:license did not match any files."

(inspite of matching string). Also suggester returns no result during that time.

**Reply by vladak** [reply]
Anything interesting in the webapp logs when the empty search result was returned ?

**Reply by jkjha** [reply]
Sorry, I didn't catch webapp logs... I will try to reproduce this and update here.
